docs/enhancements/image-service-cloud-authentication.md

```diff
@@ -0,0 +1,164 @@
+
+This security definition will apply only to the following endpoints:
+
+* `/v2/infra-envs/{infra_env_id}/downloads/files`
+* `/v2/infra-envs/{infra_env_id}/downloads/minimal-initrd`
+
+These are the endpoints that the image service uses to fetch image customization information
+and are the only ones that are needed for this process.
+
+The secret and expiration date will be stored as columns in the infraEnv database table
+and will not be accessible through the infraEnv API. The expiration will remain the same (4 hours)
+
+A new API endpoint will be added:
+
+* `/v2/infra-envs/{infra_env_id}/downloads/refresh-image-secret`
+
+This new endpoint and the infraEnv GET/PATCH/POST will be protected by user credential
+authentication and authorization.
+
+Both the assisted service and image service will accept a new URL parameter `image_secret`
+to be used with this authentication method.
+
+### Risks and Mitigations
+
+Validating authentication against a database has a performance impact compared to JWT
+validation which doesn't need to contact a separate system. That said, we still already
+check the database for authorization and the added time to check that the correct secret
+is provided should be small compared to the total download time so it likely won't be
+something users notice.
+
+Introducing a new authentication system always involves some risk that it will be implemented
+incorrectly, but this addition should be easy enough to understand and address any issues that
+could come up.
+
+## Design Details [optional]
+
+### Open Questions
+
+* Should we switch the on-prem behavior to use the same auth strategy?
+  - Currently when we deploy using the operator we also create a public/private key pair
+    which are used to create auth tokens.
+  - These tokens don't expire and are primarily used to authenticate communication from
+    the agent to assisted service.
+* Should we also support RHSSO user tokens in a header for these endpoints?
+  - The image service could also pass a header through the assisted service
+
+### UI Impact
+
+### Test Plan
+
+- Image downloads from the cloud with the image service and the new authentication work correctly.
+- An image can not be downloaded without a secret or with an incorrect secret.
+- The refresh image secret endpoint invalidates the old token (requests no longer succeed with the old URL) and replaces the URL with a valid one
+- URLs with an existing token expire after 4 hours
+
+## Drawbacks
+
+Implementing our own authentication system is generally risky. We already have a system
+(RHSSO) that works in the cloud, gives us authorization as well, and we don't maintain.
+The only reason we can't use this system for this use case is that we want to make
+the download easier for users. The most secure, and most flexible option would be
+to require an OCM JWT token for this download, just as we do for every other API call.
```

Comment:
This paragraph summarizes my main concern. I would really prefer it if we didn't create another authentication system. 

Have we asked the RHSSO folks what they think? Maybe they know of a way for us to accomplish what we need. It be nice to have their feedback as part of this proposal.